pkg/cloudprovider/servicecontroller/servicecontroller.go

```diff
@@ -225,44 +225,25 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 		glog.Infof("LB already exists and doesn't need update for service %s", namespacedName)
 		return nil, notRetryable
 	}
-	if cachedService != nil {
-		// If the service already exists but needs to be updated, delete it so that
-		// we can recreate it cleanly.
-		if cachedService.Spec.CreateExternalLoadBalancer {
-			glog.Infof("Deleting existing load balancer for service %s that needs an updated load balancer.", namespacedName)
-			if err := s.ensureLBDeleted(cachedService); err != nil {
-				return err, retryable
-			}
-		}
-	} else {
-		// If we don't have any cached memory of the load balancer, we have to ask
-		// the cloud provider for what it knows about it.
-		endpoint, exists, err := s.balancer.GetTCPLoadBalancer(s.loadBalancerName(service), s.zone.Region)
+
+	if !service.Spec.CreateExternalLoadBalancer {
+		_, exists, err := s.balancer.GetTCPLoadBalancer(s.loadBalancerName(service), s.zone.Region)
 		if err != nil {
 			return fmt.Errorf("Error getting LB for service %s", namespacedName), retryable
 		}
-		if exists && stringSlicesEqual(service.Spec.PublicIPs, []string{endpoint}) {
```

Comment:
Getting rid of the endpoint comparison for non-cached clusters will also reduce correctness in cases where we don't have a service cached and the existing LB for the service is out of sync with what k8s knows about the service.